.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+permissions:
+  contents: read
+
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners
+# GitHub Actions does not support Docker, PostgreSQL server on Windows, macOS :(
+
+concurrency:
+  # On master/release, we don't want any jobs cancelled so the sha is used to name the group
+  # On PR branches, we cancel the job if new commits are pushed
+  # More info: https://stackoverflow.com/a/68422069/253468
+  group: ${{ github.ref == 'refs/heads/trunk' && format('ci-main-{0}', github.sha) || format('ci-main-{0}', github.ref) }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: 'Java 8'
+    runs-on: ubuntu-latest
+    steps:
+    - name: 'Checkout xalan-java'
+      uses: actions/checkout@v3
+    - name: 'Set up JDK 8'
+      uses: actions/setup-java@v2
+      with:
+        distribution: zulu
+        java-version: 8
+    - name: 'Build Xalan jars'
+      run: |
+        ant jar
+    - uses: actions/checkout@v3
+      name: 'Checkout xalan-test'
+      with:
+        repository: apache/xalan-test
+        path: xalan-test
+        ref: xalan-j_2_7_x
+    - name: 'Run xalan-test tests'
+      working-directory: xalan-test
```

Comment:
The problem is seen already after merging #9 into `xalan-j_2_7_1_maint`.  I suggest it should be fixed there first.  This PR would get updated automatically then.

Reply:
AFAIK PRs can't me modified after merging

Reply:
If the tests work fine when we invoke them as per our usual process, and don't work fine under CI, then CI is invoking them incorrectly. Yes, I am going to propose moving the tests into xalan-java so it's a more standard maven package and the default CI test setup should then work, but that's not yet on the table. We should either make CI replicate our current development environment well enough to run the regression tests as they are designed to be invoked, or turn off CI again until that can be done. 

During development, xalan-test is currently a separate project cloned as a sibling of xalan-java.
In our old distribution jars, xalan-test was shipped as a subdirectory of xalan-java.
Our development test drivers set up paths which should allow tests to be executed from either location. 

I'd need to look at exactly what CI is doing, but @stanio 's analysis sounds entirely plausible.

And, yes, this gets fixed by a new PR, just like any other fix of code after it has been checked in. Nothing unusual about it, other than that we ideally should have tested better before accepting it for merge. Bugs happen. Bugs get fixed. Business as usual.

Reply:
> I've suggested/meant fixing it on the `xalan-j_2_7_1_maint` branch by means of additional PR/commit.

@vlsi, care to provide a fix for your original contribution?  I don't have experience with Github Actions, but I've tried the second option (adjusting the final build command) works in my fork: stanio/xalan-java@a62bf319d0e01a0d30b15e35d3ac9a593fcb4dc0.  I haven't been able to make it check out `xala-java` and `xalan-test` as sibling directories – is it possible with Github Actions?

Reply:
> I haven't been able to make it check out `xala-java` and `xalan-test` as sibling directories

I've managed it, but I'm a bit puzzled about some details: stanio/xalan-java@5b21fa28619a19bfb0b66a19f79c4b8aa24bece8.  Not sure if this is better than adjusting just the final build command.

Reply:
> I've managed it, but I'm a bit puzzled about some details

O.k.  On the `xalan-j_2_7_x` branch [the configuration differs](https://github.com/apache/xalan-test/blob/b2abcc438dd926fe18d175bd5efe981b8832cb41/build.xml#L93) from `master`:

```xml
    <property name="xalan.relpath" value="../java"/>
```

Reply:
@stanio, @jkesselm, if you feel confident, please fix it.
I provided the major part of the CI scripts, and I expect adjusting the command should be trivial for committers/PMCs as I assume they use tests. I do not want spend my time on reverse-engineering scripts that will be deleted soon (both ant and xalan-test)

Reply:
Fix applied directly to the branch, since I was iterating in GitHub to test the changes (and to figure out what it's test environment would let me do -- couldn't check out in parallel, which fulldist apparently currently requires, but could check out and then move to that location.)

The smoketest-after-build instructions _are_ in the xalan-java README, y'know. The only real issue with those instructions is that the xsltc.conf target, unlike the interpretive, doesn't currently have a "report only regressions" equivalent -- so for it specific failures are "correct" and that isn't compatible with the CI framework. For now I've simply dropped those tests from CI; we can reintroduce them, and perhaps expand the smoketest suite in other ways, as that is addressed and/or known divergences from the spec are resolved.

Merge up to master should now be safe.

Reply:
> This looks like checking out `xalan-test` as a subdirectory of `xalan-java`, while the build script is expecting it to be a sibling directory:

Actually, tests can be run from either location. But fulldist needs it as a sibling.

Unfortunately the CI system didn't like me checking out into Xalan's parent directory. But it was willing to let me check out locally and then _move_ the tests up to a sibling position.

Since I didn't discover the fulldist requirement until after I'd gotten past test, the yaml script currently checks out xalan-test as a child, runs the tests from there, then moves it up to sibling before running fulldist. This may not be the most self-evident approach, and the yaml script now has a comment explaining this. 

Yes, setting test.relpath was probably at least as clean. But I wanted fulldist to replicate the environment we use for our own production builds.

And yes, we could try the mv workaround earlier rather than having test in different places at different times. 

"Make it work, make it good, make it great. It now demonstrably works. Clean up can be left as technical debt, though I'd give it low priority since this will change somewhat if test is brought into the main Xalan tree.